crates/node/primitives/src/client/application.rs

```diff
@@ -57,6 +55,32 @@ impl NodeClient {
         Ok(Some(bytes))
     }
 
+    pub async fn get_precompiled_application_bytes(
+        &self,
+        application_id: &ApplicationId,
+    ) -> eyre::Result<Option<Arc<[u8]>>> {
+        let handle = self.datastore.handle();
+
+        let key = key::ApplicationMeta::new(*application_id);
+
+        let Some(application) = handle.get(&key)? else {
+            return Ok(None);
+        };
+
+        if application.precompiled_blob.blob_id() == BlobId::from([0; 32]) {
+            return Ok(None);
+        }
+
+        let Some(bytes) = self
+            .get_blob_bytes(&application.precompiled_blob.blob_id())
+            .await?
+        else {
+            bail!("fatal: application points to dangling precompiled blob");
+        };
+
+        Ok(Some(bytes))
+    }
```

Comment:
in retrospect, we should never need this function.. the same way we never use `get_application_bytes`